extensions/steve/steve.go

```diff
@@ -63,20 +62,53 @@ func WaitForSteveResourceCreation(client *rancher.Client, v1Resource *v1.SteveAP
 	return err
 }
 
-// CreateResource creates a steve resource and polls the resulting object until it comes up active.
-func CreateAndWaitForResource(client *rancher.Client, v1ResourceType string, v1Resource any, poll bool, interval, timeout time.Duration) (*v1.SteveAPIObject, error) {
-	resource, err := client.Steve.SteveType(v1ResourceType).Create(v1Resource)
+// CreateAndWaitForResource creates a steve resource and polls the resulting object until it is in the desired state.
+func CreateAndWaitForResource(client *rancher.Client, clusterID, v1ResourceType string, v1Resource any, state string, interval, timeout time.Duration) (*v1.SteveAPIObject, error) {
+	steveClient, err := GetDownstreamClusterClient(client, clusterID)
 	if err != nil {
 		return nil, err
 	}
+
+	resource, err := steveClient.SteveType(v1ResourceType).Create(v1Resource)
+	if err != nil {
+		return nil, err
+	}
+
 	logrus.Infof("Creating %s(%s)", resource.Kind, resource.Name)
 
-	if poll {
-		err := WaitForSteveResourceCreation(client, resource, interval, timeout)
+	if interval != 0 && timeout != 0 {
+		err := WaitForResourceState(steveClient, resource, state, interval, timeout)
 		if err != nil {
 			return resource, err
 		}
 	}
 
 	return resource, nil
 }
+
+// GetDownstreamClusterClient fetches the the client of a downstream cluster
+func GetDownstreamClusterClient(client *rancher.Client, clusterID string) (*v1.Client, error) {
```

Comment:
since we are creating a client for both downstream cluster as well as local cluster, can the function name be changed to something generic like GetClusterClient()?